Modules/Core/Controllers/View.php

```
@@ -2,214 +2,297 @@
+     * @legacy-function generateInvoicePdf
+     * @legacy-file application/modules/guest/controllers/View.php
      */
-    public function generateInvoicePdf($invoice_url_key, $stream = true, $invoice_template = null)
+    public function generateInvoicePdf(string $invoice_url_key, bool $stream = true, ?string $invoice_template = null): void
     {
-        $invoice = (new InvoicesService())->guestVisible()->where('invoice_url_key', $invoice_url_key)->get();
-        if ($invoice->numRows() == 1) {
-            $invoice = $invoice->row();
-            if ( ! $invoice_template) {
-// TODO: Laravel autoloads helpers - $this->load->helper('template');
+        $invoice = $this->invoiceService->getByUrlKey($invoice_url_key);
```

PHPMD (2.15.0)

The variable $invoice_url_key is not named in camelCase. (undefined)

(CamelCaseVariableName)